### ISSUE TITLE
Sharing buttons: Sharing buttons settings from Jetpack do not apply in Calypso

### ISSUE THREAD

**Issue by katinthehatsite** [Automattic/jetpack#12419]
We had two reports this week that sharing buttons that were selected to show under posts in Gutenberg - Jetpack settings were not showing under the post. After checking, it seems that these settings did not apply in Calypso when you switch to Classic editor. I was not able to reproduce this on my end with customer's themes but we had two people reporting it with different themes. 

#### Steps to reproduce
1. Create a post and select Jetpack settings to show sharing buttons under the post:

<img width="873" alt="Screen Shot 2019-05-16 at 9 04 58 AM" src="https://user-images.githubusercontent.com/25575134/57855882-b7ee1780-77b9-11e9-8b42-96d4d29fe899.png">

2. Publish a post 

#### What I expected
To see sharing buttons under individual posts

#### What happened instead
Sharing buttons were not showing

#### Next Steps:

When I switch to classic editor and checked Sharing options there, the box for the button was unchecked: 
<img width="646" alt="Screen Shot 2019-05-16 at 9 07 05 AM" src="https://user-images.githubusercontent.com/25575134/57855981-fe437680-77b9-11e9-8027-7134cb4463cf.png">

After enabling that button, the buttons came through under the posts. 

#### Details
- We had two customers reporting this with themes Dyad 2 and Baskerville 2
- On both sites, the site language is set to French
- I was not able to reproduce this with the same set of sharing buttons and same settings on my test site but the issue is still occurring on customer sites: https://parlons2tout.wordpress.com/ and https://wonderbrunette.com/

#### Screenshot / Video
Reports: 
2021524-zen
2004878-zen

**Comment by simison** [comment]
I'll transfer this to Jetpack repo since that's where the feature is developed — including the one  visible through Calypso.

**Comment by kraftbj** [comment]
Tagging @codebykat since she worked on this most recently for her expert opinions.

@katinthehatsite Can you confirm if this was occurring only on Simple sites or also on Atomic/Jetpack sites?

**Comment by codebykat** [comment]
I believe we do intentionally disable sharing buttons on private posts, yes. (Which makes sense to me, although it's kind of annoying that you can still choose the option in the UI.) On scheduled posts I wouldn't be surprised if they didn't show in preview, but showed once the post was published.

I think the "Switch to Classic Editor" here is a red herring -- the same behavior would likely occur if you checked the boxes in the block editor and then simply reloaded the page. Probably the setting is disabled by the backend on publish/update and that doesn't propagate back to the UI.

**Comment by katinthehatsite** [comment]
@kraftbj The issue so far has been reported for two Simple sites. I did not come across Atomic sites. I am also not able to reproduce it on my end but we saw it happening on two sites (by the way, apologies, I assigned you by mistake to this issue when I meant to respond to your comment).

**Comment by katinthehatsite** [comment]
@codebykat One of the posts on which it happened was https://parlons2tout.wordpress.com/2019/04/26/parlons2-anderson-paak-make-it-better/ (to give a specific example). 
- What basically happened is that the user had the option to show the sharing buttons enabled in Jetpack settings in Gutenberg. 
- The post was not set to private when it was published and once it happened, the buttons were not there.
-  When I switched back to classic, I saw that the option was unchecked under Sharing to show buttons but it was still checked in Jetpack. 
- After I enabled them in Classic editor and saved, they finally showed up. 
- Basically, I am not sure why the buttons are not showing initially on public posts when the setting to show them was selected in Jetpack. 
- I tried to reproduce it on my end with exactly the same setup as our customers have but I was not able to see the issue. 

I am not sure what exactly they are doing to trigger it but we can contact one of them for specific steps. Would this be helpful?

**Comment by ccwalburn** [comment]
I have another example of something similar happening on this AT site: https://proteinveggiecarb.com/

They have been given a workaround to publish the post and then go back in and toggle the sharing buttons on and off, and that seems to work, but the user is looking for a fix so that the buttons will display when the post is published.

We had them try publishing from different browsers and also had them deactivate Yoast but neither of those worked. This is the latest post where the sharing buttons did not display.
https://proteinveggiecarb.com/nut-butter-drizzle/

2043181-zen

**Comment by jburciaga0** [comment]
To further add to this, I think there is a general display bug that I cannot find a separate GitHub for but seems related enough to post here:

This is the second time I've seen this @codebykat but the first time I got screenshots for it.

Sometimes, rarely, when a customer posts a new post, the sharing buttons don't show up at all:
https://d.pr/i/qd1TZK

but when I go into that post and then use the Jetpack buttons to toggle off and toggle on the sharing buttons, it works:
https://d.pr/i/d0VvIm

Again the steps are:
1. Identify sharing buttons are not showing on latest post
2. Go into SU or SSP onto customer's site
3. Go to the Jetpack [button/menu](https://d.pr/i/XEpMT8)
4. Toggle off "Show Sharing buttons", click "update"
5. Toggle on "Show Sharing buttons", click "update"
6. Profit. The sharing buttons show up like in the second screenshot above.

This is the second time this has happened exactly and the site is a Simple site, just like the one previous. Thank you!

**Comment by jeherve** [comment]
@jburciaga0 If you can, it'd be good to figure out what editor was used to publish the posts in the first place (old Calypso, Calypso block editor, wp-admin, something else), and if you can get a series of steps the author followed to create the post (did they save the draft or publish right away, did they schedule, ...). Basically, anything that we could use to try to get into that initial wrong state after publication.

Thank you!

**Comment by donalirl** [comment]
In #2126257-zen we have a case where the user uses a workaround to publish the post, uncheck the show sharing buttons box, update post, re-check the show sharing buttons box, and then update the post again.

They create a post at My Site > Site > Posts, and use the block editor. They create a template post, which they copy, update, and publish, or else save as a draft, and post later. They do not schedule the posts.

**Comment by bikedorkjon** [comment]
I believe I'm seeing this problem in 2125360-zen going to ask them to check their wp-admin to see if they are selected.

**Comment by katinthehatsite** [comment]
I have seen another case with this post: https://cantabricus.blog/2019/04/18/el-bosque-de-notre-dame-y-las-restauraciones/ in 2248623-zen

Here is a short screencast: 

![Sharing](https://user-images.githubusercontent.com/25575134/62427020-90bd2d00-b6ba-11e9-8e7f-1c1c999f8b1f.gif)
After reselecting these buttons from the Calypso, the buttons started showing.

**Comment by thebud15** [comment]
This also happened with #15994722-hc

Turning the sharing buttons off and then on within the Sharing section of the post caused them to come back up, but no word as to the cause of this.

**Comment by aisajib** [comment]
Just a note that I had a user who had sharing buttons turned on for a page (atomic site with the block editor) but they weren't showing up. Repeating what @thebud15 did above (turning it off and back on from the Sharing section of the page editor) appears to have fixed it. Reference: #16296204-hc

**Comment by SiobhyB** [comment]
This came up again in 16557684-hc (followed up in 2482262-zen). 

It's an Atomic site that's using Gutenberg. Toggling the settings off and on is the workaround, as with other reports here. The user would like to get to the bottom of why the icons aren't displaying correctly when posts are published, though.

**Comment by gwensmithx** [comment]
There's another issue of this here: 6000060-hc and followup 2484996-zd. Clicking on the Publicize settings within each individual post and toggling the "Show Sharing Buttons" setting on and off solves the issue but it's horribly annoying to have to go into each individual post to do this as the initial setting is ignored.

**Comment by jerrysarcastic** [comment]
Also reported by the owner of  `keskesink.com` in #16823013-hc and `mommycatharsis.com` in #16511650-hc (same owner, so two of their sites are affected)

toggling the sharing settings (per post) in wp-admin causes the sharing buttons to return, as others  have noted above

**Comment by codebykat** [comment]
I can't dive back into this but for those who are debugging, it might be helpful to look at the postmeta on a post in the broken state to determine whether Gutenberg is initially setting the flags incorrectly.

**Comment by katinthehatsite** [comment]
Another report in #2620575-zen

**Comment by jeherve** [comment]
It should be okay to close that one since we moved away from the Calypso editor. Closing.
